File: app/src/main/java/rod/bailey/trafficatnsw/hazard/filter/AdmitRegionalHazardFilter.java
package rod.bailey.trafficatnsw.hazard.filter;

import java.util.List;

import rod.bailey.trafficatnsw.json.hazard.XHazard;
import rod.bailey.trafficatnsw.json.hazard.XRegion;
import rod.bailey.trafficatnsw.json.hazard.XRoad;
import rod.bailey.trafficatnsw.util.MLog;

public class AdmitRegionalHazardFilter implements IHazardFilter {

	private static final String TAG = AdmitRegionalHazardFilter.class
			.getSimpleName();

	@Override
	public boolean admit(XHazard hazard) {
		assert hazard != null;

		boolean result = false;

		List<XRoad> roads = hazard.getRoads();
		if ((roads != null) && (!roads.isEmpty())) {
			XRoad road = roads.get(0);
			String roadRegionStr = road.getRegion();

			if (roadRegionStr != null) {
				try {
					XRegion roadRegion = XRegion.valueOf(roadRegionStr);
					if (roadRegion != null) {
						result = roadRegion.isRegional();
					}
				} catch (Throwable thr) {
					MLog.w(TAG,
							"Found a region string that could not be decoded to an enum element: "
									+ roadRegionStr, thr);
				}
			}
		}

		return result;
	}

}

File: app/src/main/java/rod/bailey/trafficatnsw/hazard/filter/AdmitRegionalHazardFilter.kt
package rod.bailey.trafficatnsw.hazard.filter

import rod.bailey.trafficatnsw.hazard.data.XHazard
import rod.bailey.trafficatnsw.hazard.data.XProperties
import rod.bailey.trafficatnsw.hazard.data.XRegion

class AdmitRegionalHazardFilter : IHazardFilter {
	override fun admit(hazard: XHazard): Boolean {
		val props: XProperties? = hazard.properties
		var result = false

		if (props != null) {
			val roads =props.roads
			if ((roads != null) && !roads.isEmpty()) {
				val road = roads[0]
				val roadRegionStr: String? = road.region
				if (roadRegionStr != null) {
					val roadRegion: XRegion? = XRegion.valueOf(roadRegionStr)
					result = roadRegion?.isRegional ?: false
				}
			}
		}

		return result
	}

	companion object {
		private val TAG = AdmitRegionalHazardFilter::class.java.simpleName
	}
}
